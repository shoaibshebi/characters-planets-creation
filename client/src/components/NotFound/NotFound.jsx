import React from "react";
import { Grid, Typography } from "@mui/material";

import classes from "./NotFound.module.scss";
import planetLoaderIcon from "../../assets/images/planet-loader.svg";

export default function NotFound({ text }) {
  return (
    <Grid container className={classes.container}>
      <Typography variant="h1" className={classes.code}>
        404
      </Typography>
      <img alt="planet icon" src={planetLoaderIcon} />
      <Typography variant="h6" className={classes.title}>
        {text}
      </Typography>
    </Grid>
  );
}
