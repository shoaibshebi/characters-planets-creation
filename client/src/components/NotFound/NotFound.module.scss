.container {
  justify-content: center;
  align-items: center;
  flex-direction: column !important;
}
.code {
  font-weight: 800 !important;
  font-size: 8rem !important;
}
.title {
  font-family: "Poppins" !important;
  font-size: 32px !important;
  font-weight: 800 !important;
  line-height: 48px !important;
}
