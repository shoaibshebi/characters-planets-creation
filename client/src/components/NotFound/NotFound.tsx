import React from "react";
import PropTypes from "prop-types";
import { Grid, Typography } from "@mui/material";

import classes from "./NotFound.module.scss";
import planetLoaderIcon from "../../assets/images/planet-loader.svg";

interface Props {
  text: string;
  code: number | string;
}

const NotFound: React.FC<Props> = ({ text, code }) => {
  return (
    <Grid container className={classes.container}>
      <Typography variant="h1" className={classes.code}>
        {code}
      </Typography>
      <img alt="planet icon" src={planetLoaderIcon} />
      <Typography variant="h6" className={classes.title}>
        {text}
      </Typography>
    </Grid>
  );
};

NotFound.propTypes = {
  text: PropTypes.string,
  code: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
};

export default NotFound;
