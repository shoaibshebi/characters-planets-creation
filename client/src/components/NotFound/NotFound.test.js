import { render, screen, cleanup } from "@testing-library/react";

import NotFound from "./NotFound.jsx";

const text = "It seems like you are lost in the space.";

afterEach(cleanup);
it("matches snapshot", () => {
  const { asFragment } = render(<NotFound text={text} />);

  expect(asFragment()).toMatchSnapshot();
});
