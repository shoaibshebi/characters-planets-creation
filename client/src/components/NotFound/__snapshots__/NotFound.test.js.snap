// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root MuiGrid-container container css-11lq3yg-MuiGrid-root"
  >
    <h1
      class="MuiTypography-root MuiTypography-h1 code css-o2w69a-MuiTypography-root"
    />
    <img
      alt="planet icon"
      src="planet-loader.svg"
    />
    <h6
      class="MuiTypography-root MuiTypography-h6 title css-2ulfj5-MuiTypography-root"
    >
      It seems like you are lost in the space.
    </h6>
  </div>
</DocumentFragment>
`;
